P2P data platform proposal (#1866)
w3f/Grants-Program: data_platform_with_deep_indexed_data_and_staking_reports.md

> a# Polkadot Data platform with deep indexed data and staking reports

Reviewer: ```suggestion
# Polkadot Data platform with deep indexed data and staking reports
```

> # Data platform proposal for Web3 fund
> 

Reviewer: ```suggestion
```

According the web3 fund need from RFP Polkadot ecosystem needs:

- extract on-chain data and transform it to be available for SQL querying by data analysts
- provide public access through query engine to build data sets on-top of indexed data
- provide UI interface for building dashboards on-top of data sets to visualize data insights for decision makers

We saw some teams that already building such solutions for the RFP and want to describe our key features which could provide the best user experience according to our vision:

- modular architecture allows you
    - to use any indexer such as Subsquid, Subquery or custom indexer (Mbelt in [P2P.org](http://P2P.org) case)
    - to use any database as a DWH for storing public data (in current RFP we will use BigQuery as DWH)
    - to customize ETL process for any indexer/DWH solution via Airflow DAGs
    - to customize the solution for real-time data enrichment process via direct streaming to DWH
- dashboard with sli/slo for data quality of ETL pipeline - we provide transparent and clear data quality metrics to estimate the quality of ETL
- [P2P.org](http://P2P.org) infrastructure (P2P RPC for indexation)

### Potential users

> We define the users of the platform as data people who generates insights from the data to support data-driven base for decision makers. This people has

Reviewer 2: ```suggestion
We define the users of the platform as data people who generates insights from the data to support data-driven base for decision makers. These people have:
```

- tech skills to write SQL queries
- basic statistics to define calculate and estimate metrics for the business
- expertise of the particular domain (such as NFT, particular deFi)

Current state of Dune platform (as the Leader of the market) is about 20k users. Dune is focusing on Ethereum data and DeFi in Ethereum ecosystem. Current number of active wallets in Ethereum is 400k. So, we see that current state of the crypto industry demonstrates a high proportion of data-people to all token holders with is 5%. 

We believe that in case of mass adoption we will not see this proportion but we need to involve more people still. Our global vision for this problem is to provide ability to work with the data without SQL knowledge. That is why we develop Kolmo - AI assistant.

![](https://github.com/bigslimdogg/web3_pic/blob/3a0406565507ada022e57797adf8258edf1e50e3/ai_assistant.png)

> Our concern that we can attract first users from Polkadot ecosystem which hasn’t been covered by the current data-platforms. After that we are going to provide innovative opportunity to enter in data-driven decisions without SQL skills

Reviewer 2: ```suggestion
Our concern is that we can first attract users from the Polkadot ecosystem, which hasn’t been covered by the current data-platforms. After that we are going to provide an innovative opportunity to enter in data-driven decisions without SQL skills.
```

- Staking SLI: 100% (no problems detected)

Kusama and Moonbeam are mostly similar.

Moonriver sli:

- Block processing mean time: 914 msec (min - 81 msec, max- 98 sec)
- Mean latency between block in RPC node to block in database: ~ 95 сек (min - 12 sec, max 240 sec)
- Blocks processed correctly in first attempt (others are fixed automatically) ~ 96%
- Staking SLI: 95%

We also have an SLI [dashboard here](https://reports.p2p.org/superset/dashboard/124/?native_filters_key=hx3xwFAW7LaoWw6vWbB6eY7JD3BOjObbcgnwy_BkExJw20nkHu2NtfTYLkDe22Kl), but thresholds used in this dashboard were set by internal team and could be different for final public solution.

## Team :busts_in_silhouette:

### Team members

- Developers:
    - Ivan Safronov (developer of Mbelt)
    - Artem Kiselev (team lead of indexation team in P2P.org)
- DWH engineers:
    - Kovalev German - (data engineer)
    - Kamalov Sergey (team lead of DWH team in P2P.org)
- Product owner: Kotenko Nick

### Team Code Repos

Indexer: [Mbelt](https://github.com/p2p-org/polkadot-profit-transformer/tree/mbelt3/prod)

Public DWH: “p2p-data-platform” in bigQuery you can find in public access though bigQ console

### Team's experience

> We believe that proposed product can brought not only new user experience for validators and NO managers but provide truly decentralization with the performance improvement. P2P.org has long-term experience in more then 25 blockchains since 2018. We can bring the high level expertise to help new participants perform better by providing recommendations on-top of the core solution.

Reviewer 2: ```suggestion
We believe that proposed product can bring not only a new user experience for validators and NO managers but also provide true decentralization with the performance improvement. P2P.org has long-term experience in more then 25 blockchains since 2018. We can bring the high level expertise to help new participants perform better by providing recommendations on-top of the core solution.
```
